src/webgpu/api/operation/rendering/draw.spec.ts

```diff
@@ -44,7 +28,330 @@ TODO:
   - mode= {draw, drawIndexed, drawIndirect, drawIndexedIndirect}
 `;
 
+import { params, pbool, poptions } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/framework/util/util.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+g.test('arguments')
+  .desc(
+    `Test that draw arguments are passed correctly by drawing triangles in a grid.
+Horizontally across the screen are triangles with increasing "primitive id".
+Vertically down the screen are triangles with increasing instance id.
+Increasing the |first| param should skip some of the beginning triangles on the horizontal axis.
+Increasing the |first_instance| param should skip of the beginning triangles on the vertical axis.
+The vertex buffer contains two sets of disjoint triangles, and base_vertex is used to select the second set.
+The test checks that the center of all of the expected triangles is drawn, and the others are empty.
+The fragment shader also writes out to a storage buffer. If the draw is zero-sized, check that no value is written.
+
+Params:
+  - first= {0, 3} - either the firstVertex or firstIndex
+  - count= {0, 6} - either the vertexCount or indexCount
+  - first_instance= {0, 2}
+  - instance_count= {0, 4}
+  - indexed= {true, false}
+  - indirect= {true, false}
+  - vertex_buffer_offset= {0, 32}
+  - index_buffer_offset= {0, 16} - only for indexed draws
+  - base_vertex= {0, 9} - only for indexed draws
+  `
+  )
+  .cases(
+    params()
+      .combine(poptions('first', [0, 3] as const))
+      .combine(poptions('count', [0, 6] as const))
+      .combine(poptions('first_instance', [0, 2] as const))
+      .combine(poptions('instance_count', [0, 4] as const))
```

Comment:
Maybe also test 1 instance? 0 instance means 0 triangles will be drawn. Ditto for count.

Reply:
0 is intentional. I think @kainino0x said we don't need to be testing 1 vs. non-zero here?

Reply:
Eh, added the cases for 1 triangle and 1 instance.